src/org/ods/orchestration/usecase/JiraUseCase.groovy

```diff
@@ -497,33 +499,50 @@ class JiraUseCase {
         return jira.getIssuesForJQLQuery(jqlQuery) ?: []
     }
 
-    List createSecurityVulnerabilityIssues(List aquaCriticalVulnerabilityRepos) {
+    Set createSecurityVulnerabilityIssues(List aquaCriticalVulnerabilityRepos) {
         if (!jira) {
             logger.warn("createSecurityVulnerabilityIssues: Could *NOT* create security vulnerability issues " +
                 "because jira has invalid value.")
             return []
         }
-        def securityVulnerabilityIssueKeys = [];
+        Set securityVulnerabilityIssueKeys = [];
         for (def repo : aquaCriticalVulnerabilityRepos) {
             def jiraComponentId = getJiraComponentId(repo)
             for (def vulnerability : repo.data.openshift.aquaCriticalVulnerability) {
-                def vulerabilityMap = vulnerability as Map
+                def vulnerabilityMap = vulnerability as Map
                 def issueKey = createOrUpdateSecurityVulnerabilityIssue(
-                    vulerabilityMap.name,
+                    vulnerabilityMap.name,
+                    vulnerabilityMap.solution,
                     jiraComponentId,
                     buildSecurityVulnerabilityIssueDescription(
-                        vulerabilityMap,
+                        vulnerabilityMap,
                         repo.data.openshift.gitUrl,
                         repo.data.openshift.gitBranch,
                         repo.data.openshift.repoName,
                         repo.data.openshift.nexusReportLink))
+                logger.debug("Create/update security vulnerability issue with key ${issueKey} " +
+                    "for ${vulnerabilityMap.name}, component ${jiraComponentId}, " +
+                    "repo ${repo.data.openshift.repoName}")
                 securityVulnerabilityIssueKeys.add(issueKey)
             }
         }
         return securityVulnerabilityIssueKeys
     }
 
-    String createOrUpdateSecurityVulnerabilityIssue(String vulnerabilityName, String jiraComponentId,
+    void checkIssueConstainsSolutionAndUpdateIfNeeeded(Map issue, String solution) {
+        String issueDescription = issue.fields.description as String
+        if (issueDescription.contains(solution)) {
```

Comment:
Change the condition to avoid an empty return: if(!....) {}

Reply:
Done.